jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -648,12 +648,15 @@ private ByteBuffer acquireBuffer()
         return _aggregate;
     }
 
-    private void releaseBuffer()
+    private void releaseBuffer(Throwable failure)
     {
         if (_aggregate != null)
         {
             ByteBufferPool bufferPool = _channel.getConnector().getByteBufferPool();
-            bufferPool.release(_aggregate);
+            if (failure == null)
+                bufferPool.release(_aggregate);
+            else
+                bufferPool.remove(_aggregate);
```

Comment:
Can we make this behaviour conditional on version >= 2.0 ? Otherwise we open a simple h1 DOS vector that will allow the buffer pool to be cycled too easily.

The real fix is #12727

Reply:
We can't backport https://github.com/jetty/jetty.project/pull/12727 to 10.0.x/11.0.x without a huge effort and these releases are EOL, so adding a simple extra if (httpVersion < 2) then always release looks reasonable.

Let me add that.